libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -0,0 +1,1929 @@
+                    vector_store.distance_strategy,
+                    params,
+                )
+
+        else:
+            await _acreate_hnsw_index(
+                connection,
+                vector_store.table_name,
+                vector_store.distance_strategy,
+                params,
+            )
+
+    await _handle_context(client, context)
+    return
+
+
+@_handle_exceptions
+async def _acreate_hnsw_index(
+    connection: AsyncConnection,
+    table_name: str,
+    distance_strategy: DistanceStrategy,
+    params: Optional[dict[str, Any]] = None,
+) -> None:
+    idx_name, ddl = _get_hnsw_index_ddl(table_name, distance_strategy, params)
+
+    # check if the index exists
+    if not await _aindex_exists(connection, idx_name):
+        with connection.cursor() as cursor:
+            await cursor.execute(ddl)
+            logger.info("Index created successfully...")
+    else:
+        logger.info("Index already exists...")
+
+
+@_handle_exceptions
+async def _acreate_ivf_index(
+    connection: AsyncConnection,
+    table_name: str,
+    distance_strategy: DistanceStrategy,
+    params: Optional[dict[str, Any]] = None,
+) -> None:
+    idx_name, ddl = _get_ivf_index_ddl(table_name, distance_strategy, params)
+
+    # check if the index exists
+    if not await _aindex_exists(connection, idx_name):
+        with connection.cursor() as cursor:
+            await cursor.execute(ddl)
+        logger.info("Index created successfully...")
+    else:
+        logger.info("Index already exists...")
+
+
+@_handle_exceptions
+def drop_table_purge(client: Any, table_name: str) -> None:
+    """Drop a table and purge it from the database.
+
+    Args:
+        client: The OracleDB connection object.
+        table_name: The name of the table to drop.
+
+    Raises:
+        RuntimeError: If an error occurs while dropping the table.
+    """
+    connection = _get_connection(client)
+    if connection is None:
+        raise ValueError("Failed to acquire a connection.")
+    if _table_exists(connection, table_name):
+        with connection.cursor() as cursor:
+            ddl = f"DROP TABLE {table_name} PURGE"
+            cursor.execute(ddl)
+        logger.info("Table dropped successfully...")
+    else:
+        logger.info("Table not found...")
+    return
+
+
+@_handle_exceptions
+async def adrop_table_purge(client: Any, table_name: str) -> None:
+    """Drop a table and purge it from the database.
+
+    Args:
+        client: The OracleDB connection object.
+        table_name: The name of the table to drop.
+
+    Raises:
+        RuntimeError: If an error occurs while dropping the table.
+    """
+
+    async def context(connection: Any) -> None:
+        if await _atable_exists(connection, table_name):
+            with connection.cursor() as cursor:
+                ddl = f"DROP TABLE {table_name} PURGE"
+                await cursor.execute(ddl)
+            logger.info("Table dropped successfully...")
```

Comment:
Logging would be better if the table name was included.  Or is this a security risk?